js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx

```diff
@@ -155,43 +369,186 @@ const RightPanelDetail = ({
   );
 };
 
-const MiddlePanel = () => {
+const MiddlePanel = ({
+  assetKey,
+  selectedEvaluationId,
+  maxMaterializationsPerMinute,
+}: {
+  assetKey: Omit<AssetKey, '__typename'>;
+  selectedEvaluationId?: string;
+  maxMaterializationsPerMinute: number;
+}) => {
+  const {data, loading, error} = useQuery<GetEvaluationsQuery, GetEvaluationsQueryVariables>(
+    GET_EVALUATIONS_QUERY,
+    {
+      variables: {
+        assetKey,
+        cursor: selectedEvaluationId,
+        limit: 1,
+      },
+    },
+  );
+
+  const evaluationData = React.useMemo(() => {
+    return data?.autoMaterializeAssetEvaluations[0];
+  }, [data?.autoMaterializeAssetEvaluations]);
+
+  const conditionResults = React.useMemo(() => {
+    const results: Partial<{
+      materializationIsMissing: boolean;
+      codeVersionHasChangedSinceLastMaterialization: boolean;
+      upstreamCodeVersionHasChangedSinceLastMaterialization: boolean;
+      upstreamDataHasChangedSinceLatestMaterialization: boolean;
+      requiredToMeetAFreshnessPolicy: boolean;
+      requiredToMeetADownstreamFreshnessPolicy: boolean;
+
+      // skip conditions
+      waitingOnUpstreamData: boolean;
+      exceedsXMaterializationsPerHour: boolean;
+    }> = {};
+    evaluationData?.conditions.forEach((cond) => {
+      switch (cond.__typename) {
+        case 'DownstreamFreshnessAutoMaterializeCondition':
+          results.requiredToMeetADownstreamFreshnessPolicy = true;
+          break;
+        case 'FreshnessAutoMaterializeCondition':
+          results.requiredToMeetAFreshnessPolicy = true;
+          break;
+        case 'MissingAutoMaterializeCondition':
+          results.materializationIsMissing = true;
+          break;
+        case 'ParentMaterializedAutoMaterializeCondition':
+          results.upstreamDataHasChangedSinceLatestMaterialization = true;
+          break;
+        case 'ParentOutdatedAutoMaterializeCondition':
+          results.waitingOnUpstreamData = true;
+          break;
+        case 'MaxMaterializationsExceededAutoMaterializeCondition':
+          results.exceedsXMaterializationsPerHour = true;
+          break;
+        default:
+          console.error('Unexpected condition', (cond as any).__typename);
+          break;
```

Comment:
This seems fine but a bit verbose, another way we might do this is say `requiredToMeetAFreshnessPolicy = conditions.some(c => c.__typename === 'FreshnessAutoMaterializeCondition')`

Doing it that way might be nice because each row component could take the evaluationData and the typename it's related to and do that check on it's own?

Reply:
I just get itchy when I think of all the extra loops we'd be doing but its nbd